tapdb/assets_store.go

```diff
@@ -1212,20 +1217,32 @@ func (a *AssetStore) FetchAssetProofs(ctx context.Context,
 func (a *AssetStore) FetchProof(ctx context.Context,
 	locator proof.Locator) (proof.Blob, error) {
 
-	// We don't need anything else but the script key since we have an
-	// on-disk index for all proofs we store.
-	scriptKey := locator.ScriptKey
+	// We have an on-disk index for all proofs we store, so we can use the
+	// script key as the primary identifier.
+	args := FetchAssetProof{
+		TweakedScriptKey: locator.ScriptKey.SerializeCompressed(),
+	}
+
+	// But script keys aren't unique, so if the locator explicitly specifies
+	// an outpoint, we'll use that as well.
+	if locator.OutPoint != nil {
```

Comment:
Do we need to _enforce_ this now? Eg: what happens if the script key is still used, but we have two blobs to return, which do we select to return back to the caller?

Reply:
We will enforce this in a later commit, yes. Or rather, _if_ there are multiple proofs for just the script key, we return an error and require the user to also specify the outpoint.